### DISCUSSION TITLE
Architecture for separating the features of the meta-index from the features of the package repository

### DISCUSSION THREAD

**Discussion by tchoutri** [flora-pm/flora-server#472] [Questions]
It is now becoming vital to separate the features enabled when we are in "meta-index" mode from the ones that must be enabled when in "package repository" mode.

## Meta-index characteristics

* Import of foreign package repositories (hackage, cardano, etc)
* Fetching metadata from said repositories
* Fetching README and CHANGELOG from the source repositories
* User account creation disabled

## Package repository characteristics

* Tarball upload is manual
* Fetching README and CHANGELOG from the uploaded tarball
* Namespace is unique, and configured at first start-up.

[To be completed]

**Comment by RaoulHC** [comment]
I'm not convinced about the namespace being unique, having provenance is useful, especially if a deployment ends up mirroring packages from a couple of different sources (on that note, the list of packages we mirror from different sources should be configurable, I'll write up an issue for that). But in package repository mode flora would have ensure there are no clashes and provide a view of tarballs without namespaces that cabal can understand.

At least while there isn't support for namespaces, it would be nice for cabal/hackage to support them in some form eventually.
